src/crimson/net/Errors.h

```diff
@@ -43,17 +37,12 @@ inline std::error_code make_error_code(error e)
   return {static_cast<int>(e), net_category()};
 }
 
-inline std::error_condition make_error_condition(error e)
-{
-  return {static_cast<int>(e), net_category()};
-}
-
 } // namespace crimson::net
 
 namespace std {
 
-/// enables implicit conversion to std::error_condition
+/// enables implicit conversion to std::error_code and std::error_condition
```

Comment:
> /// enables implicit conversion to std::error_code and std::error_condition

This comment is mainly from https://en.cppreference.com/w/cpp/error/error_code/is_error_code_enum _"This template may be specialized for a user-defined type to indicate that the type is eligible for std::error_code and std::error_condition automatic conversions."_. Maybe the reference is not strictly correct this time.

Except the comment, I think we can still define `struct is_error_code_enum<crimson::net::error> : public true_type {};` for the purpose to compare error code as conditions. See the definition and usage of io_errc: https://en.cppreference.com/w/cpp/io/io_errc.

Reply:
> Except the comment, I think we can still define `struct is_error_code_enum<crimson::net::error> : public true_type {};` for the purpose to compare error code as conditions. See the definition and usage of io_errc: https://en.cppreference.com/w/cpp/io/io_errc.

i am more interested in "why".

Reply:
> I think we can still define `struct is_error_code_enum<crimson::net::error> : public true_type {};` for the purpose to compare error code as conditions. See the definition and usage of io_errc: https://en.cppreference.com/w/cpp/io/io_errc.

again, explicit template specialization of std::is_error_code_enum only allows implicit conversion from the error code to std::error_code.

Reply:
> i am more interested in "why". std::error_code is for platform/system dependent representations of error, while std::error_condition is for platform/system independent ones.

My understanding is that our error codes defined in (simplified) "crimson::net::error" are ceph network layer dependent errors.

>  i just don't understand why we need to define error "code" instead of "condition" here.

IMHO because firstly "`crimson::net::protocol_aborted` and the like" are error codes we are used to generate errors. It is a simplification to also use them as "condition" in if-clauses, just like io_errc.